source/common/upstream/eds.cc

```diff
@@ -22,16 +24,21 @@ EdsClusterImpl::EdsClusterImpl(const Json::Object& config, Runtime::Loader& runt
 
 void EdsClusterImpl::initialize() { subscription_->start({cluster_name_}, *this); }
 
-bool EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
+void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
   std::vector<HostSharedPtr> new_hosts;
   if (resources.size() != 1) {
-    ENVOY_LOG(warn, "Unexpected EDS resource length: {}", resources.size());
-    return false;
+    // We need to allow server startup to continue, even if we have a bad
+    // config.
+    runInitializeCallbackIfAny();
```

Comment:
This is still really fragile (having to call runInitializeCallbackIfAny()) before throwing. In the old code, we had onFetchComplete() which runs no matter. Can we maybe add onConfigUpdateComplete() or something like that here to do catch-all logic? I think that would be better.